### DISCUSSION TITLE
Require mail from payee on per-app basis

### DISCUSSION THREAD

**Discussion by ghost** [btcpayserver/btcpayserver#2316] [Ideas / Feature Requests]
Currently, the requirement that the payee has to provide a mail address is set on a per-store basis. Meaning that all apps inside one store have the same email requirement setting.

If the admin wants to set different mail requirements for different apps, then the only way to do this is by setting up a new store, with new on-chain and lightning wallet. This is not reasonable, especially if the lightning node is already well established. [EDIT: actually LN node can easily be reused across stores. I think my argument for this feature request still holds.]

I would like to set the mail requirement on a per-app basis, so that different apps in the same store can have different user input.

A potential use case, is that in one store, the admin has a app with payments from users whom he would like to reach out to after payment confirmation, to send for example a ticket etc. While the other app is a donation page where the admin does not care to know sent the money.

Of course, a user can just type in `foo@mail.com` if he does not want to be contacted, but this is one additional UX hurdle, which can be annoying for example donations.

**Comment by bolatovumar** [comment]
This sounds reasonable for me. It makes sense to allow this setting to be configured on a per-app basis rather than per store basis. I guess the store could have the default setting and each store could modify it if needed? 

I would like to know if anyone has reasons for why this would not be a good idea.

**Comment by Kixunil** [comment]
FYI, you can reuse same LN node in a different store. It's completely safe and trivial.
I'm not sure about xpub or even hot wallets though. (Is there a risk of generating same address twice? Will PayJoin work?)
